x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/querydsl/query/TermsQuery.java

```diff
@@ -11,23 +11,28 @@
 import org.elasticsearch.xpack.sql.tree.Location;
 import org.elasticsearch.xpack.sql.type.DataType;
 
+import java.util.Collections;
 import java.util.LinkedHashSet;
 import java.util.List;
 import java.util.Objects;
+import java.util.Set;
 
 import static org.elasticsearch.index.query.QueryBuilders.termsQuery;
 
 public class TermsQuery extends LeafQuery {
 
     private final String term;
-    private final LinkedHashSet<Object> values;
+    private final Set<Object> values;
 
     public TermsQuery(Location location, String term, List<Expression> values) {
         super(location);
         this.term = term;
         values.removeIf(e -> e.dataType() == DataType.NULL);
-        this.values = new LinkedHashSet<>(Foldables.valuesOf(values, values.get(0).dataType()));
-        this.values.removeIf(Objects::isNull);
+        if (values.isEmpty()) {
+            this.values = Collections.emptySet();
+        } else {
+            this.values = new LinkedHashSet<>(Foldables.valuesOf(values, values.get(0).dataType()));
```

Comment:
I'm assuming that here all the values in `values` are of the same type otherwise we would have had the error message about different types.

Reply:
yes that's true. I can even use the type from the left value.